### PR TITLE
Fix recipes from shipdist

### PR DESCRIPTION
**Pull request by siscia**
The bulk of this PR is adding system checks for several packages.

Those checks are very simple, this carry a little bit of risk, but is a quite non-fragile approach and provides a simple way to hack into the build system.

The PR also fix a few inconsistencies regarding FairRoot.

### CODE REVIEW

**Comment by olantwin** [comment]
I don't understand the ROOT system check. Could you explain why this is better than using e.g. `root-config --features` and/or `root-config --version`

**Comment by siscia** [comment]
It does not require root-config in the path.

And it is very similar to all the other checks.

**Comment by olantwin** [comment]
I'm not convinced by the other checks of the format `ls foo/bar > /dev/null` either. We should at least check the version, ideally also features of all of those packages.

`root-config` not being available isn't a problem. If `root-config` isn't in the path, I'd say ROOT is misconfigured and unuseable.

Also, we shouldn't assume that the `$BLA_ROOT` environment variables are set. They usually aren't configured for system versions of e.g. ROOT.

**Comment by siscia** [comment]
Please, feel free to do this better :)

**Comment by olantwin** [comment]
I'd love to, but I have 0 time right now.

**Comment by siscia** [comment]
Sorry, I was a little nervous.

I definitely understand your point, but often time this choices are a matter of pragmatism and useful user interface.

While it is definitely attractive to pin the dependencies of every libraries we use it comes with some costs.

First of all the cost of checking those versions, not all the libraries have a simple and clean way to check their version. Moreover we have few lines of bash script to make those checks happens and doing complex things will make them unreadable (hence nobody will be able to check them or to modify then).

Another cost is inflexibility, we will definitely want to have the possibility to modify the library we use for compilation, in this way is simple, if we pin the version it will be much more complex.

Moreover, this way is very very simple, whoever can just open the file and understand what is going on.

Of course, there is the possibility that some builds pick up the wrong version of some libraries, but how likely it is?

You need to pass a specific flag to alibuild to use those checks `--always-prefer-system` and if you pass that check you know what you are doing.

Moreover I expect that these checks are used mostly after having sourced the environment, that will set all the variables to the correct one from CVMFS.

I expect that in 99% of cases this solution will just work as intended and honestly I expect that in the other few cases it won't matter.

Moreover I believe that pinning all the dependencies, while safer, will make everything a little more complex in all cases. A flexible system is invaluable in my experience.

To recap, what Oliver is suggesting is completely right, myself wanted to do something similar, but after some work I figured out that is not a good solution for the long term.

In most of the case this solution will work just fine giving us enough flexibility and safety, for the other cases I believe is not worth, neither the implementation effort nor the actual result.

Cheers,

Oliver, sorry if I sounded mean, I was just nervous for other reasons :)

**Comment by ThomasRuf** [comment]
Looks very reasonable to me.

**Comment by siscia** [comment]
@olantwin did I buy you over this?

What you think?

**Comment by olantwin** [comment]
@siscia Sorry for the late reply. I was on holiday.

Do I understand correctly this is supposed to be only used on lxplus and by advanced users and is not supposed to become the default?

In that case this seems like a reasonable approach to take.

**Comment by siscia** [comment]
Very good!

**Comment by siscia** [comment]
@ThomasRuf  This PR contains also the 2018 frozen version.
